apix-rest-v1/src/main/java/eu/xenit/apix/rest/v1/nodes/NodesWebscript1.java

```diff
@@ -588,60 +588,68 @@ public void retrieveAncestors(@UriVariable String space, @UriVariable String sto
+                            NodeRef nodeRef;
+                            NodeRef copyFrom = null;
+                            if (createNodeOptions.copyFrom == null) {
+                                nodeRef = nodeService
+                                        .createNode(parent, createNodeOptions.name,
+                                                new QName(createNodeOptions.type));
+                            } else {
+                                copyFrom = new NodeRef(createNodeOptions.copyFrom);
+                                if (!nodeService.exists(copyFrom)) {
+                                    response.setStatus(HttpStatus.SC_NOT_FOUND);
+                                    writeJsonResponse(response, "CopyFrom does not exist");
+                                    return null;
+                                }
+                                nodeRef = nodeService.copyNode(copyFrom, parent, true);
+                            }
 
-                        return nodeRef;
-                    }
-                }, false, true);
+                            MetadataChanges metadataChanges;
+                            QName type;
+                            if (createNodeOptions.type != null) {
+                                type = new QName(createNodeOptions.type);
+                            } else if (createNodeOptions.type == null && createNodeOptions.copyFrom != null) {
+                                type = nodeService.getMetadata(copyFrom).type;
+                            } else {
+                                response.setStatus(HttpStatus.SC_BAD_REQUEST);
+                                writeJsonResponse(response,
+                                        "Please provide parameter \"type\" when creating a new node");
+                                return null;
+                            }
+                            metadataChanges = new MetadataChanges(type, null, null,
+                                    createNodeOptions.properties);
+                            nodeService.setMetadata(nodeRef, metadataChanges);
 
-        NodeRef resultRef = new NodeRef(resultObject.toString());
+                            return nodeRef;
+                        }
+                    }, false, true);
 
-        NodeInfo nodeInfo = this
-                .nodeRefToNodeInfo(resultRef, this.fileFolderService, this.nodeService, this.permissionService);
+            NodeRef resultRef = new NodeRef(resultObject.toString());
 
-        writeJsonResponse(response, nodeInfo);
+            NodeInfo nodeInfo = this
+                    .nodeRefToNodeInfo(resultRef, this.fileFolderService, this.nodeService, this.permissionService);
+
+            writeJsonResponse(response, nodeInfo);
+        } catch (org.alfresco.service.cmr.model.FileExistsException fileExistsException) {
+            FileExistsException wrappingException = new FileExistsException(fileExistsException);
```

Comment:
These assignments (648-650) should be part of the wrapping FileExistsException constructor. Otherwise every place where we use this, will have to copy-paste these same assignments.